### PR TITLE
move WQuickXdgOutputManager to protocols/WXdgOutput

### PR DESCRIPTION
**Pull request by Dami-star**
Changed from inheriting from WQuickWaylandServerInterface to inheriting from WServerInterface.

### CODE REVIEW

File: src/server/protocols/wxdgoutput.cpp [hunk 1]
```diff
@@ -423,26 +422,36 @@ void WQuickXdgOutputManager::setLayout(WOutputLayout *layout)
...
+void WXdgOutputManager::create(WServer *wserver)
+{
+    Q_UNUSED(wserver)
+    W_D(WXdgOutputManager);
     if (d->layout) {
         d->manager = way_xdg_output_manager_v1_create(server()->handle()->handle(),
                                                       d->layout->handle(),
                                                       d->scaleOverride);
-        return new WServerInterface(d->manager, d->manager->global);
+        new WServerInterface(d->manager, d->manager->global);
```
Comment (zccrs) [inline src/server/protocols/wxdgoutput.cpp]: 不要，让 WXdgOutputManager 继承 WServerInterface 就行

File: examples/tinywl/main.cpp [hunk 2]
```diff
@@ -55,11 +56,20 @@ Helper::Helper(QObject *parent)
...
+    , m_xwayland_xdgoutputmanager(new WXdgOutputManager())
 {
     m_seat->setEventFilter(this);
     m_seat->setCursor(m_cursor);
     m_cursor->setThemeName(getenv("XCURSOR_THEME"));
     m_cursor->setLayout(m_outputLayout);
+
+    m_xwayland_xdgoutputmanager->setLayout(m_outputLayout);
+    m_xwayland_xdgoutputmanager->setScaleOverride(1.0);
+    m_xdgoutputmanager->setTargetClients(m_xwayland_xdgoutputmanager->targetClients(),false);
```
Comment (zccrs) [inline examples/tinywl/main.cpp]: , 后加空格

File: src/server/qtquick/private/wquickxdgoutput_p.h [hunk 3]
```diff
@@ -1,39 +0,0 @@
-// Copyright (C) 2024 Lu YaNing <luyaning@uniontech.org>.
```
Comment (zccrs) [inline src/server/qtquick/private/wquickxdgoutput_p.h:1]: 没有对应的  cpp 文件要删除吗

**Review by zccrs — CHANGES_REQUESTED** [review]

File: examples/tinywl/main.cpp (hunk 2 above)
Comment (Dami-star) [reply examples/tinywl/main.cpp:69]: donw

Comment (Dami-star) [reply examples/tinywl/main.cpp]: done

File: src/server/qtquick/private/wquickxdgoutput_p.h (hunk 3 above)
Comment (Dami-star) [reply src/server/qtquick/private/wquickxdgoutput_p.h:1]: 修改文件名提升移动文件了

File: src/server/protocols/wxdgoutput.cpp (hunk 1 above)
Comment (Dami-star) [reply src/server/protocols/wxdgoutput.cpp]: done

File: src/server/protocols/wxdgoutput.cpp [hunk 4]
```diff
@@ -423,26 +422,35 @@ void WQuickXdgOutputManager::setLayout(WOutputLayout *layout)
...
+    auto handle = nativeInterface<QWXdgOutputManagerV1>();
+    return handle->handle()->global;
+}
+
+void WXdgOutputManager::create(WServer *wserver)
+{
+    Q_UNUSED(wserver)
+    W_D(WXdgOutputManager);
     if (d->layout) {
         d->manager = way_xdg_output_manager_v1_create(server()->handle()->handle(),
```
Comment (zccrs) [inline src/server/protocols/wxdgoutput.cpp:448]: m_handle = xxxxx

**Review by zccrs — CHANGES_REQUESTED** [review]

**Pull request closed without merging**